src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -91,6 +91,74 @@ public class TestExtensions
     /// </summary>
     internal bool AreDataCollectorsCached { get; set; }
 
+    /// <summary>
+    /// Merge two extension maps.
+    /// </summary>
+    /// 
+    /// <param name="extensionMap1">First extension map.</param>
+    /// <param name="extensionMap2">Second extension map.</param>
+    /// 
+    /// <returns>
+    /// A map representing the merger between the two input maps.
+    /// </returns>
+    internal static IDictionary<string, ISet<string>> MergeExtensionMaps(
+        IDictionary<string, ISet<string>> extensionMap1,
+        IDictionary<string, ISet<string>> extensionMap2)
+    {
+        var isExtensionMap1Valid = (extensionMap1 != null && extensionMap1.Count > 0);
+        var isExtensionMap2Valid = (extensionMap2 != null && extensionMap2.Count > 0);
+
+        // Sanity checks.
+        if (!isExtensionMap1Valid && !isExtensionMap2Valid) return new Dictionary<string, ISet<string>>();
```

Comment:
I would prefer not to mix styles (i.e. either parenthesis for 1 line ifs or not). Inside of the foreach you do use parenthesis.

Usually in the code, it's the option with the parenthesis which is favored.

Reply:
Done, thanks.